src/baseline_bots/bots/smart_order_accepter_bot.py

```diff
@@ -287,6 +290,102 @@ def generate_support_proposals(self, comms_obj: MessagesData) -> Dict[str, str]:
                 final_messages[recipient] = str(suggested_proposals)
                 comms_obj.add_message(recipient, str(suggested_proposals))
         return final_messages
+        
+    def is_order_aggressive_to_powers(self, order: str, powers: List[str]):
+        """
+        check if the order is aggressive by
+        1. to attack allies unit
+        2. to move to allies' SC
+        3. support attack allies unit
+        4. support move to allies' SC
+
+        :param order: an order as string, e.g. "A BUD S F TRI"
+        :param powers: powers that we want to check if a bot is having aggressive move to
+        :return: Boolean
+
+        """
+        order_token = get_order_tokens(order)
+        if order_token[0][0] == "A" or order_token[0][0] == "F":
+            # for 1 and 2
+            if order_token[1][0] == "-":
+                # get location - add order_token[0] ('A' or 'F') at front to check if it collides with other powers' units
+                order_unit = order_token[0][0] + order_token[1][1:]
+                for power in powers:
+                    if self.power_name != power:
+                        # if the order is to attack allies' units
+                        if order_unit in self.game.powers[power].units:
+                            return True
+                        # if the order is a move to allies' SC
+                        if order_token[1][2:] in self.game.powers[power].centers:
+                            return True
+            # for 3 and 4
+            if order_token[1][0] == "S":
+                # if support hold
+                if len(order_token) == 3:  # ['A BUD', 'S', 'A VIE']
+                    return False
+                order_unit = order_token[2][0] + order_token[3][1:]
+                for power in powers:
+                    if self.power_name != power:
+                        # if the order is to attack allies' units
+                        if order_unit in self.game.powers[power].units:
+                            return True
+                        # if the order is a move to allies' SC
+                        if order_token[3][2:] in self.game.powers[power].centers:
+                            return True
+        return False
+
+    @gen.coroutine
+    def get_non_aggressive_order(self, order: str, powers: List[str]):
+        """
+        return a non-aggressive order with other options in dipnet beam order, if none left, support its own unit. if none around, support self hold.
+
+        :param order: an order as string, e.g. "A BUD S F TRI"
+        :param powers: powers that we want to check if a bot is having aggressive move to
```

Comment:
reword this parameter description. smtg like "powers that we want to check if a bot is being aggressive towards"

Reply:
ok